### PR TITLE
Add card readers page business url field client validation

### LINKED ISSUE #3512: Remove the feature flag _wcpay_feature_card_readers
Related to #1440

Card Readers is a new, upcoming page that will:

* list all readers linked to an account
* allow customizing reader receipt branding/settings

A basic code implementation of the page was recently added, but the page itself is currently hidden in the WP Admin frontend until it is tested well.

Once the page is fully ready, its feature flag `_wcpay_feature_card_readers` should be removed to make the page visible (owing to restrictions imposed by the `card_present_eligible` flag).

### LINKED ISSUE #3952: Client-side validations for terminal receipt settings
As noted in #3874, in certain cases our server-side validations may prevent saving the overall plugin settings. To fix this, a possible solution is to move validations to client-side (browser) and ensure proper sanitization still happens on the server.

This issue is to explore and implement this or another similar approach.

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #3952

#### Changes proposed in this Pull Request
Adding a client-side validation for the Business URL setting field on the Card Readers settings page.

Screenshot 2022-03-21 at 17 55 57

#### Testing instructions
- Update the `_wcpay_feature_card_readers` option to `1`.
- Update the `card_present_eligible` to `true`.
- Navigate to `Payments` -> `Card Readers page` -> `Receipt details`.
- Enter invalid URL address in Business URL field. 
- Verify error is displayed and Save button is disabled.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/card-readers/settings/sections/business-details.js [hunk 1]
```diff
@@ -22,6 +22,23 @@ const BusinessDetailsSection = () => {
 		setAccountBusinessURL,
 	] = useAccountBusinessURL();
 
+	useLayoutEffect( () => {
+		//businessUrlRef.blur();
```
Comment (kalessil) [inline client/card-readers/settings/sections/business-details.js:26]: Dead code?

File: changelog/add-card-readers-inline-validation [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: add
+Comment: Add card readers business URL field browser validation
+
+
```
Comment (kalessil) [inline changelog/add-card-readers-inline-validation:5]: The file formatting is slightly confusing, it's typically like [this](https://github.com/Automattic/woocommerce-payments/blob/develop/changelog/update-3861-ipp-exclude-l2-l3-data).

**Review by kalessil — CHANGES_REQUESTED** [review]
Overall looks good, left a couple of comments sound and will test after those are addressed.

File: client/card-readers/settings/sections/business-details.js (hunk 1 above)
Comment (mordeth) [reply client/card-readers/settings/sections/business-details.js]: Removed. 😊

File: changelog/add-card-readers-inline-validation (hunk 2 above)
Comment (mordeth) [reply changelog/add-card-readers-inline-validation:6]: Updated!

**Review by mgascam — APPROVED** [review]
LGTM!

File: client/card-readers/settings/index.tsx [hunk 3]
```diff
@@ -44,21 +45,26 @@ const ReadersSettingsDescription = (): JSX.Element => (
...
+	const [ isSaveDisabled, setDisabled ] = useState( false );
+
 	return (
 		<SettingsLayout displayBanner={ false }>
 			<SettingsSection description={ ReadersSettingsDescription }>
 				<LoadableSettingsSection numLines={ 20 }>
 					<Card className="card-readers-settings__wrapper">
 						<CardBody>
-							<BusinessDetailsSection />
+							<BusinessDetailsSection
```
Comment (anu-rock) [inline client/card-readers/settings/index.tsx:56]: Functionally, I understand the flow and props here. But the naming is a bit confusing. At first glance, it's hard to correlate the business details section's `isDisabled` with the save button's `isSaveDisabled`.

Does the following sound better?

```suggestion
							<BusinessDetailsSection
								setSaveDisabled={ setDisabled }
							/>
```

... and create an internal state in `BusinessDetailsSection` to show/hide the error message:

```javascript
const [ hasError, setHasError ] = useState( false );

const validateBusinessURL = ( event ) => {
		if ( event.target.checkValidity() ) {
            setHasError( true );
			setSaveDisabled( false );
		} else {
            setHasError( true );
			setSaveDisabled( true );
		}
	};
```

**Review by anu-rock — APPROVED** [review]
Overall, great work here!

I've left a nitpick comment that if addressed might improve the readability a bit.

**Review by kalessil — APPROVED** [review]
LGTM!

File: client/card-readers/settings/index.tsx (hunk 3 above)
Comment (kalessil) [reply client/card-readers/settings/index.tsx:56]: Marking as ready to merge, but it would be great to look into this comment @mordeth .

Comment (mordeth) [reply client/card-readers/settings/index.tsx:56]: Thank you, @anu-rock, your suggestion makes the readability much better. Updated.

**Pull request merged**